reverse_image_search_module/test/test_multicrop.py

```diff
@@ -0,0 +1,149 @@
+import os
+import numpy as np
+import random
+from PIL import Image
+from ..search_image import find_index_from_image, find_file_name, find_repeating_index
+import logging
+import matplotlib.pyplot as plt
+from datetime import datetime
+
+
+logging.basicConfig(level=logging.INFO)
+mylogger = logging.getLogger()
+
+
+DATA_IMAGES_PATH = 'data/img/full/'
+K_CROPPING = 10
+N_SEARCH = 10
+IMAGE_MAX_NUMBER = 5000
+
+
+def random_cropping(image_width, image_height, min_size, K):
+    for _ in range(K):
+        x_ini = random.randint(0, image_width - min_size)
+        y_ini = random.randint(0, image_height - min_size)
+
+        x_end = random.randint(x_ini + min_size, image_width)
+        y_end = random.randint(y_ini + min_size, image_height)
+
+        yield x_ini, y_ini, x_end, y_end
+
+
+def get_image_matching_many(image_names, many=False):
+    positive = 0
+
+    for _im in image_names:
+        try:
+            image = Image.open(os.path.join(DATA_IMAGES_PATH, _im))
+            # to check the size and truncation
+            _ = image.resize((255, 255))
+        except Exception as ex:
+            mylogger.info(f'image {_im} load error')
+            mylogger.info(str(ex))
+            continue
+        
+        
+        idxs, dists = [], []
+
+        for x_ini, y_ini, x_end, y_end in random_cropping(255,255, 200, K_CROPPING):
```

Comment:
Random croppings here may lead to non-reproduced behavior: like if I’m submitting the same image to art bot twice and getting different results. Pros: if the image wasn’;t found once, it can be found on the second run (works in other direction as well). Cons: can be hard to debug a specific situation and see why we got some results - this won’t be reproducible.

Reply:
solved in commit f07d64c :)